# Regresión Lineal Múltiple

1. Determinal las variables independientes
2. Determinar la variable dependiente
3. Obtener el valor de intesección

In [6]:
import pandas as pd
import numpy as np
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error, r2_score

In [7]:
#importar datos
df = pd.read_csv("Regresión múltiple.csv")
df.head()

Commissions  Calls  Driven
0           22    139    2371
1           13    132    2226
2           33    144    2731
3           38    142    3351
4           23    142    2289

### 1. Determinar las variables independientes
x1 = calls, x2 = driven

In [8]:
X = df[['Calls','Driven']]

### 2. Determinar las variables dependientes Y
y = commissions

In [9]:
Y = df[['Commissions']]

In [10]:
#Ajustar el model de regresión multiple
model = LinearRegression()
model.fit(X, Y) #Entrena el modelo
y_pred = model.predict(X)
#y_pred

#Métricas estadísticas
mse = mean_squared_error(Y, y_pred)
r2 = r2_score(Y, y_pred)

### 3. Valor de intersección

In [11]:
model.intercept_

array([-101.31444968])

In [12]:
#coeficientes
print('Coeficientes: ', model.coef_)
print("Error cuadrático medio ", mse)
print("Varianza: ", r2)

Coeficientes:  [[0.63233256 0.01568925]]
Error cuadrático medio  16.313412727936065
Varianza:  0.8352686374291524


Con base en el análisis de regresión múltiple, se concluye que las comisiones de los agentes dependen casi exclusivamente del número de llamadas realizadas (coeficiente de 0.63), mientras que las millas recorridas no representan un factor significativo (coeficiente de 0.01). No existe una interacción relevante entre ambas variables, por lo que la estrategia más eficiente para incrementar comisiones es enfocarse en el volumen de llamadas antes que en los desplazamientos físicos.